### PR TITLE
Create triage-process.md

### PR DESCRIPTION
**Pull request by jparsana**
Bug triage process definition

### CODE REVIEW

File: triage-process.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+1. Define and publish a process to handle issues in a structured manner.
+    1. Issue Handling Process
+    1. Issue Stages
+    1. Roles
+    1. SLA
+
+## Process Definition
+
+### Issue Handling Process
+1. All incoming issues and PR’s should be as. These are items that a Triager would look at on a daily basis
```
Comment (c24t) [inline triage-process.md]: "be as", assigned?

File: triage-process.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+1. Every issue triaged should be assessed for its workability. If assessing the workability of the issue requires more information, the Triager would collaborate with the OP until workability is determined.
+    1. Workability Classification:
+        1. New Feature Request
+        1. Enhancement to Existing Feature
+        1. Bug
+1. Workable Issues should be prioritized and Labeled as one of 3 priorities
+    * P0
+    * P1
+    * P2
+1. Prioritized Issues (and unworked) issues should be labeled as ‘Ready for Work’. This should signal to any contributor willing to help that they can work on this issue.
```
Comment (rghetia) [inline triage-process.md]: OR assign label 'Won't fix' if the issue is not workable.

File: triage-process.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+1. Once a contributor is working on this issue, the issue should be:
+    1. Assigned to him
+    1. Labeled as ‘In Progress’
```
Comment (rghetia) [inline triage-process.md]: no 'In Progress' label is not required. If it is assigned then it is in progress.

File: triage-process.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        1. New Feature Request
+        1. Enhancement to Existing Feature
+        1. Bug
+1. Workable Issues should be prioritized and Labeled as one of 3 priorities
+    * P0
+    * P1
+    * P2
+1. Prioritized Issues (and unworked) issues should be labeled as ‘Ready for Work’. This should signal to any contributor willing to help that they can work on this issue.
+1. Once a contributor is working on this issue, the issue should be:
+    1. Assigned to him
```
Comment (rghetia) [inline triage-process.md]: s/him/them/ (Gender neutral ;-)

File: triage-process.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    1. Labeled as ‘In Progress’
+1. Once work is complete and issue has been resolved it should be labeled as ‘Completed’ and Issue should be closed.
```
Comment (rghetia) [inline triage-process.md]: 'Completed' label is not required. If the issue is closed and there is a commit associated with it then it is completed.

File: triage-process.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+1. All incoming issues and PR’s should be as. These are items that a Triager would look at on a daily basis
+1. Every issue triaged should be assessed for its workability. If assessing the workability of the issue requires more information, the Triager would collaborate with the OP until workability is determined.
+    1. Workability Classification:
+        1. New Feature Request
+        1. Enhancement to Existing Feature
+        1. Bug
+1. Workable Issues should be prioritized and Labeled as one of 3 priorities
+    * P0
+    * P1
+    * P2
```
Comment (rghetia) [inline triage-process.md]: define what these priorities mean.

File: triage-process.md (hunk 1 above)
Comment (dinooliva) [reply triage-process.md]: or assessed?

File: triage-process.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+* __Reviewer__: 
+    * Person(s) whose Approval is needed to merge the PR
+* __Contributing Reviewer__: 
+    * Person who provides feedback on an issue/PR but who Approval is not necessarily needed to merge the PR.
+
+### SLAs
+* __Triager__: Daily (Business Days) review of untriaged issues with initial feedback to OP or issue transitioned to ‘Ready to Work’ within 3 days of last activity.
+* __Reviewer / Contributing Reviewer__: Provide feedback within 3 days of the last activity on a PR
+* __Contributor__: 3 day updates as comments on the PR to indicate activity
+
```
Comment (rghetia) [inline triage-process.md]: Add SLA for each priority.

File: triage-process.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+## Process Definition
+
+### Issue Handling Process
+1. All incoming issues and PR’s should be as. These are items that a Triager would look at on a daily basis
+1. Every issue triaged should be assessed for its workability. If assessing the workability of the issue requires more information, the Triager would collaborate with the OP until workability is determined.
+    1. Workability Classification:
+        1. New Feature Request
+        1. Enhancement to Existing Feature
+        1. Bug
```
Comment (rghetia) [inline triage-process.md]: Change it to.
1. feature-request
2. enhancement
3. bug

Above are also label names.

File: triage-process.md (hunk 1 above)
Comment (jparsana) [reply triage-process.md]: Done.

File: triage-process.md (hunk 8 above)
Comment (jparsana) [reply triage-process.md]: Done. Also added wont-fix as a label.

File: triage-process.md (hunk 6 above)
Comment (jparsana) [reply triage-process.md]: done

File: triage-process.md (hunk 2 above)
Comment (jparsana) [reply triage-process.md]: Added the wont fix label as part of workablity classification step. If the person is here(at this step) the issue is deemed workable. let me know if you see it differently.

File: triage-process.md (hunk 4 above)
Comment (jparsana) [reply triage-process.md]: done. Good catch!

File: triage-process.md (hunk 3 above)
Comment (jparsana) [reply triage-process.md]: Its not a label. I think you might be reading the column incorrectly.  See action lablel column far right in the same row.

File: triage-process.md (hunk 5 above)
Comment (jparsana) [reply triage-process.md]: Same as above - Its not a label. I think you might be reading the column incorrectly. See action lablel column far right in the same row.

File: triage-process.md (hunk 7 above)
Comment (jparsana) [reply triage-process.md]: what would be the respective SLAs the team has agreed to. If this has been agreed to in a prior discussion please post that here. In my opinon it might be too fine grained and dont think its needed at this time given where the project is at. However I will let the team make the final call.

File: triage-process.md (hunk 8 above)
Comment (dinooliva) [reply triage-process.md]: I think we decided on wontfix (no hyphen) because it is pretty standard on github.

File: triage-process.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+1. Define and publish a process to handle issues in a structured manner.
+    1. Issue Handling Process
+    1. Issue Stages
+    1. Roles
+    1. SLA
+
+## Process Definition
+
+### Issue Handling Process
+1. All incoming issues and PR’s should be assessed. These are items that a Triager would look at on a daily basis
```
Comment (c24t) [inline triage-process.md]: Nit: replace these smart quotes everywhere

File: triage-process.md (hunk 8 above)
Comment (dinooliva) [reply triage-process.md]: Additional classification: docs

File: triage-process.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+### Roles
+* __Triager__: 
+    * Person who is triaging the issue by determining its workability. This person is responsible for getting the ticket to one of two labels - 1) Ready for Work 2) Cancelled. They are responsible for triaging  by working with the OP to get additional information as needed and analyzing the issue and adding relevant details/information/guidance that would be helpful to the resolution of the issue.
+
+* __OP__: 
+    * Original Poster. This is the person who has opened or posted the issue.
+
+* __Contributor__:
+    * Person(s) that are actually doing the work related to the ticket. Once work is done they work with the Reviewer and the Contributing Reviewers to get feedback implemented and complete the work. They  are responsible for making sure issue status label is up to date (In Progress, In Review, Completed, Blocked, Cancelled)
```
Comment (mayurkale22) [inline triage-process.md]: Nit: Add comma i.e. `Once work is done, they ...`

File: triage-process.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+### Issue Stages
+Stage | Meaning | Action/Label
+------|---------|-------------
+New | Ticket is ready for triage. | Issue should be labeled ‘New’ 
+Ready for Work | Ticket triaging has been completed and is up for grabs to be picked up to be worked on by a contributor. | Ticket is labeled ‘Ready for work’. Label ‘New’ should be removed when moving to this stage.
+In Progress | Ticket is actively being worked upon by a contributor. | Contributor who has picked up this ticket shall assign the ticket to themselves.
+In Review | Implementation work related to the ticket has been completed. Work related to a Review and/or any changes requested as part of the review  is in progress.  | Issues with a PR associated with it will be considered as in ‘In Review’
+Completed | Work for this issue has been completed and reasonably tested. | No label is needed. Issue is simply closed.
+Blocked | Issue is blocked for one reason or another. Typically where the delay or waiting period is more than 5 days. | Ticket should be labelled ‘Blocked’
+Won't Fix | Issue is not workable.  | Issue should have comment indicating why its not fixable.
```
Comment (mayurkale22) [inline triage-process.md]: typo: its -> it's ?

File: triage-process.md (hunk 2 above)
Comment (rghetia) [reply triage-process.md]: SG.

File: triage-process.md (hunk 9 above)
Comment (jparsana) [reply triage-process.md]: This was done. I had copy pasted what you had shared in the gdoc.

File: triage-process.md (hunk 10 above)
Comment (jparsana) [reply triage-process.md]: done

File: triage-process.md (hunk 11 above)
Comment (jparsana) [reply triage-process.md]: Done

File: triage-process.md (hunk 3 above)
Comment (rghetia) [reply triage-process.md]: table is clear but the above line contradicts the table. If I read 'the issue should be Labeled as "In Progress"' I cannot infer anything but to apply a label called "In Progress". 
I would rephrase the above to
```
1. Once a contributor is working on this issue, the issue should be:
    1. Assigned to them. (Assigned issues are considered to be in "In Progress" stage.)
```

File: triage-process.md (hunk 5 above)
Comment (rghetia) [reply triage-process.md]: same as above. Perhaps, rephrase it to
```
Once work is complete and the issue has been resolved, simply close the Issue. A closed issue with associated merged PR is considered as "Complete".
```

File: triage-process.md (hunk 7 above)
Comment (rghetia) [reply triage-process.md]: SG.

File: triage-process.md (hunk 8 above)
Comment (jparsana) [reply triage-process.md]: Done. Also have updated the Issue Stages section. I have kept completed and blocked stages in there since nothign was mentioned explicitly. LMK if you want those removed.

Comment (dinooliva) [reply triage-process.md]: We want to keep 'blocked' as a label but 'completed' is unnecessary as a label (it is a stage but is indicated by the issue being closed).

Comment (jparsana) [reply triage-process.md]: Done. blocked is specified in the label column and completed has no label, but it is called out as a stage.

File: triage-process.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+1. Define and publish a process to handle issues in a structured manner.
+    1. Issue Handling Process
+    1. Issue Stages
+    1. Roles
+    1. SLA
+
+## Process Definition
+
+### Issue Handling Process
+1. All incoming issues and PR’s should be assessed. These are items that a Triager would look at on a daily basis
```
Comment (dinooliva) [inline triage-process.md]: We currently want this document to focus only on issues and not PR's - if PR's get to a point that they need a process, we'll develop a different spec.

File: triage-process.md (hunk 3 above)
Comment (dinooliva) [reply triage-process.md]: 'ready-for-work' and 'in-progress' should not be called labels - they're stages.

File: triage-process.md (hunk 12 above)
Comment (jparsana) [reply triage-process.md]: done

File: triage-process.md (hunk 3 above)
Comment (jparsana) [reply triage-process.md]: done

File: triage-process.md (hunk 5 above)
Comment (jparsana) [reply triage-process.md]: done.

File: triage-process.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+        1. feature-request
+        1. enhancement
+        1. bug
+        1. wontfix
+        1. docs
+1. Workable Issues should be prioritized and Labeled as one of 3 priorities
+    * P0 = Critical (Drop everything and work on it.)
+    * P1 = Important (Work on it if no P0 are present. Use judgment to prioritize items within a list of P1s)
+    * P2 = Low (Work on it when no P1 are left or some one from community wants to contribute)
+1. Prioritized and workablity classified issues  issues should are considered as ready-for-work. This should signal to any contributor willing to help that they can work on this issue.
```
Comment (dinooliva) [inline triage-process.md]: Prioritized and workablity classified issues  issues should are ->
 Prioritized and workablity classified issues are

File: triage-process.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+1. Once a contributor is working on this issue, the issue should be assigned to them. (Assigned issues are considered to be in in-progress stage.)
+1. Once work is complete and the issue has been resolved, simply close the Issue. A closed issue with associated merged PR is considered as completed.
```
Comment (dinooliva) [inline triage-process.md]: Issue(s) should be capitalized (or not) consistently.

File: triage-process.md (hunk 13 above)
Comment (jparsana) [reply triage-process.md]: done

File: triage-process.md (hunk 14 above)
Comment (jparsana) [reply triage-process.md]: done. I went thru the full doc so we should be good but PTAL incase I have missed any instance.
